applications/CompressiblePotentialFlowApplication/python_scripts/initialize_geometry.py

```diff
@@ -0,0 +1,252 @@
+
+    def Execute(self):
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','Executing Initialize Geometry')
+        self.InitializeSkinModelPart()
+        self.CalculateDistance()
+
+        ini_time=time.time()
+        while self.step < self.max_iter and self.do_remeshing:
+            self.step += 1
+            KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','##### Executing refinement #', self.step, ' #####')
+            self.ExtendDistance()
+            self.RefineMesh()
+            self.CalculateDistance()
+            self.UpdateParameters()
+        self.ModifyFinalDistance()
+        self.ExtendDistance()
+        self.CopyAndDeleteDefaultDistance()
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','Elapsed time: ',time.time()-ini_time)
+
+        ''' ############################################################################################## '''
+        ''' THIS FUNCTION CALL IS TEMPORARY AND WILL BE REMOVED ONCE THE EMBEDDED WAKE PROCESS IS DEFINED '''
+        # Find nodal neigbours util call
+        avg_elem_num = 10
+        avg_node_num = 10
+        KratosMultiphysics.FindNodalNeighboursProcess(
+            self.main_model_part, avg_elem_num, avg_node_num).Execute()
+        ''' ############################################################################################## '''
+
+    def ExecuteInitialize(self):
+        self.Execute()
+
+    def InitializeSkinModelPart(self):
+        ''' This function loads and moves the skin_model_part in the main_model_part to the desired initial point (self.initial_point).
+            It also rotates the skin model part around the initial point according to the (self.geometry_parameter)'''
+        self.skin_model_part=self.model.CreateModelPart("skin")
+        KratosMultiphysics.ModelPartIO(self.skin_model_part_name).ReadModelPart(self.skin_model_part)
+        ini_time=time.time()
+        if self.skin_model_part_name=='naca0012':
+            MoveModelPart(self.initial_point, self.skin_model_part)
+            self.origin=[0.25+self.initial_point[0],0+self.initial_point[1]]
+            angle=math.radians(-self.geometry_parameter)
+            RotateModelPart(self.origin,angle,self.skin_model_part)
+        elif self.skin_model_part_name=='ellipse':
+            MoveModelPart(self.initial_point, self.skin_model_part)
+            self.origin=[self.initial_point[0],self.initial_point[1]]
+            angle=math.radians(-self.geometry_parameter)
+            RotateModelPart(self.origin,angle,self.skin_model_part)
+        KratosMultiphysics.Logger.PrintInfo('InitializeGeometry','InitializeSkin time: ',time.time()-ini_time)
+
+    def CalculateDistance(self):
+        ''' This function calculate the distance to skin for every node in the main_model_part.'''
+        ini_time=time.time()
+        KratosMultiphysics.CalculateDistanceToSkinProcess2D(self.main_model_part, self.skin_model_part).Execute()
```

Comment:
In the function below (`ExtendDistance`) you allow 3D cases but not in this one. Is this intentionally done?

Reply:
Not intentionally, actually I should just restrict it to the 2D cases for now.